src/main/java/com/puppycrawl/tools/checkstyle/site/ClassAndSettersJavadocScraper.java

```diff
@@ -0,0 +1,155 @@
+///////////////////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code and other text files for adherence to a set of rules.
+// Copyright (C) 2001-2023 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+///////////////////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.site;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.TreeMap;
+
+import com.puppycrawl.tools.checkstyle.FileStatefulCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.DetailNode;
+import com.puppycrawl.tools.checkstyle.api.JavadocTokenTypes;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.checks.javadoc.AbstractJavadocCheck;
+import com.puppycrawl.tools.checkstyle.utils.BlockCommentPosition;
+import com.puppycrawl.tools.checkstyle.utils.CheckUtil;
+
+/**
+ * Class for scraping class javadoc and all property setter javadocs from the
+ * given checkstyle module.
+ */
+@FileStatefulCheck
+public class ClassAndSettersJavadocScraper extends AbstractJavadocCheck {
+
+    /** Name of the module being scraped. */
+    private static String moduleName;
+
+    /** The module javadoc. */
+    private static DetailNode moduleJavadoc;
+
+    /** Map of the module property setter javadocs. */
+    private static final Map<String, DetailNode> MODULE_PROPERTY_SETTER_JAVADOCS = new TreeMap<>();
+
+    /**
+     * Set the module name to be scraped.
+     *
+     * @param moduleName the module name.
+     */
+    public static void setModuleName(String moduleName) {
+        ClassAndSettersJavadocScraper.moduleName = moduleName;
+    }
```

Comment:
Can we just call this `init/initialize` and handle clearing static context and all initialization here to not put this on the caller?